### PR TITLE
home-assistant: 0.116.4 -> 0.117.0 (+ additional package updates)

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

New upstream release. Also update and clean up dependencies.

https://github.com/home-assistant/core/releases

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg test home-assistant

File: pkgs/development/python-modules/httpx/default.nix [hunk 1]
```diff
@@ -1,71 +1,61 @@
...
 , uvicorn
-, brotli
 }:
 
 buildPythonPackage rec {
   pname = "httpx";
-  version = "0.14.2";
-  disabled = isPy27;
+  version = "0.16.1";
+  disabled = pythonOlder "3.6";
```
Comment (jonringer) [inline pkgs/development/python-modules/httpx/default.nix:23]: looks like a regression:

```
builder for '/nix/store/667gj5w9jcysm7pad0b4lma5hv4zkggf-python3.7-asgi-csrf-0.7.drv' failed with exit code 2; last 10 log lines:
  ImportError while importing test module '/build/source/test_asgi_csrf.py'.
  Hint: make sure your test modules/packages have valid Python names.
  Traceback:
  test_asgi_csrf.py:7: in <module>
      from httpx._content_streams import MultipartStream
  E   ModuleNotFoundError: No module named 'httpx._content_streams'
  =========================== short test summary info ============================
  ERROR test_asgi_csrf.py
  !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
  =============================== 1 error in 0.15s ===============================
cannot build derivation '/nix/store/hda4799m3wcb5mfazk2x1s77i0v8qykr-python3.7-datasette-0.46.drv': 1 dependencies couldn't be built
```

Comment (mweinelt) [reply pkgs/development/python-modules/httpx/default.nix:23]: Probably related to https://github.com/encode/httpx/commit/354c4cac1f1796ee4d8861a4c978807398852ad5.

Comment (mweinelt) [reply pkgs/development/python-modules/httpx/default.nix:23]: Reported in https://github.com/simonw/asgi-csrf/issues/18.

Comment (jonringer) [reply pkgs/development/python-modules/httpx/default.nix:23]: looks like they imported a private module... so shame on them

**Comment by jonringer** [comment]
everything else passes:

```
https://github.com/NixOS/nixpkgs/pull/101968
4 packages failed to build:
datasette python37Packages.asgi-csrf python37Packages.datasette python38Packages.asgi-csrf

11 packages built:
google-music-scripts home-assistant python27Packages.spotipy python37Packages.google-music python37Packages.httpcore python37Packages.httpx python37Packages.spotipy python38Packages.google-music python38Packages.httpcore python38Packages.httpx python38Packages.spotipy
```

File: pkgs/development/python-modules/httpx/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/httpx/default.nix:23]: maybe just disable that test?

Doesn't look like it's failing on anything exported by the module

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 101968` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>datasette (python38Packages.datasette)</li>
    <li>google-music-scripts</li>
    <li>home-assistant</li>
    <li>python27Packages.spotipy</li>
    <li>python37Packages.asgi-csrf</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.httpcore</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.spotipy</li>
    <li>python38Packages.asgi-csrf</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.spotipy</li>
  </ul>
</details>

File: pkgs/development/python-modules/httpx/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/httpx/default.nix:23]: Disabled the tests on asgi-csrf and bumped the package to the latest version for good measure.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 101968 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>15 packages built:</summary>  <ul>    <li>datasette (python38Packages.datasette)</li>    <li>google-music-scripts</li>    <li>home-assistant</li>    <li>python27Packages.spotipy</li>    <li>python37Packages.asgi-csrf</li>    <li>python37Packages.datasette</li>    <li>python37Packages.google-music</li>    <li>python37Packages.httpcore</li>    <li>python37Packages.httpx</li>    <li>python37Packages.spotipy</li>    <li>python38Packages.asgi-csrf</li>    <li>python38Packages.google-music</li>    <li>python38Packages.httpcore</li>    <li>python38Packages.httpx</li>    <li>python38Packages.spotipy</li>  </ul></details>

**Pull request merged**